dashboard/test/models/experiment_test.rb

```diff
@@ -44,55 +51,57 @@ class ExperimentTest < ActiveSupport::TestCase
+    refute Experiment.enabled?(section: section_off, experiment_name: experiment.name)
   end
 
   test "teacher based experiment is disabled if start time is too late" do
-    create :teacher_based_experiment,
+    experiment = create :teacher_based_experiment,
       percentage: 100,
       earliest_section_at: DateTime.now + 1.day
-    section = create :section,
-      first_activity_at: DateTime.now
-    assert_empty Experiment.get_all_enabled(section: section)
+    assert_empty Experiment.get_all_enabled(section: @section)
+    refute Experiment.enabled?(section: @section, experiment_name: experiment.name)
   end
 
   test "teacher based experiment is disabled if end_time is too early" do
-    create :teacher_based_experiment,
+    experiment = create :teacher_based_experiment,
       percentage: 100,
       latest_section_at: DateTime.now - 1.day
-    section = create :section,
-      first_activity_at: DateTime.now
-    assert_empty Experiment.get_all_enabled(section: section)
+    assert_empty Experiment.get_all_enabled(section: @section)
+    refute Experiment.enabled?(section: @section, experiment_name: experiment.name)
   end
 
   test "teacher based experiment is disabled if other script assigned" do
     script = create :script
-    create :teacher_based_experiment,
+    experiment = create :teacher_based_experiment,
       percentage: 100,
       script_id: script.id + 1
-    assert_empty Experiment.get_all_enabled(section: create(:section), script: script)
+    assert_empty Experiment.get_all_enabled(section: @section, script: script)
+    refute Experiment.enabled?(section: @section, script: script, experiment_name: experiment.name)
   end
 
   test "teacher based experiment is enabled if same script assigned" do
     script = create :script
     experiment = create :teacher_based_experiment,
       percentage: 100,
       script_id: script.id
-    assert_equal [experiment], Experiment.get_all_enabled(section: create(:section), script: script)
+    assert_equal [experiment], Experiment.get_all_enabled(section: @section, script: script)
+    assert Experiment.enabled?(section: @section, script: script, experiment_name: experiment.name)
   end
 
   test "single section experiment is enabled" do
-    section = create :section
     experiment = create :single_section_experiment,
-      section_id: section.id
-    assert_equal [experiment], Experiment.get_all_enabled(section: section)
+      section_id: @section.id
+    assert_equal [experiment], Experiment.get_all_enabled(section: @section)
+    assert Experiment.enabled?(section: @section, experiment_name: experiment.name)
   end
 
   test "single section experiment is not enabled" do
-    section = create :section
-    create :single_section_experiment,
-      section_id: section.id + 1
-    assert_empty Experiment.get_all_enabled(section: section)
+    experiment = create :single_section_experiment,
+      section_id: @section.id + 1
```

Comment:
Why do we need to pass in `@section.id + 1`? Doesn't the factory automatically create a section distinct from `@section`?